File: Programmers/pro_42583_다리를지나는트럭.py
from collections import deque
def solution(bridge_length, weight, truck_weights):
	bridge = [0]*bridge_length
	que=deque()
	time, idx, cur_weight=0, 0, 0
	while(idx<len(truck_weights)):
		time+=1
		if(time>bridge_length):
			cur_weight-=que.popleft()
		if(cur_weight+truck_weights[idx]<=weight):
			cur_weight+=truck_weights[idx]
			que.append(truck_weights[idx])
			idx+=1
		else:
			que.append(0)        
	return time+bridge_length

File: Programmers/pro_42583_다리를지나는트럭.js
function solution(bridge_length, weight, truck_weights) {
  const bridge = Array(bridge_length).fill(0)
  let time = 0;
  let capacity = weight;
  while (truck_weights.length) {
    capacity += bridge.shift()
    if (truck_weights[0] <= capacity) {
      const truckW = truck_weights.shift()
      capacity -= truckW;
      bridge.push(truckW)
    } else {
      bridge.push(0)
    }
    time++;
  }
  return time + bridge_length;
}